openapi/sms/v1-review/partner-service.yaml

```diff
@@ -0,0 +1,364 @@
+servers:
+  - url: https://api.sms-partner.io/sms/partner/v1
+paths:
+  /clients/{clientId}/configuration:
+    summary: Manage a client configuration within a Partner service
+    parameters:
+      - $ref: './common.yaml#/components/parameters/clientIdPathParam'
+    put:
+      tags: [ Client Configuration ]
+      operationId: createOrUpdateClientConfiguration
+      summary: Create or update a single client's configuration
+      requestBody:
+        description: client's custom field values as a map.
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateOrUpdateClientConfigurationRequest'
+      responses:
+        '200':
+          description: Client configuration was successfully recorded within the Partner service.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CreateOrUpdateClientConfigurationResponse'
+        '400':
+            $ref: '#/components/responses/InvalidClientConfigurationFormatResponse'
+        '401':
+          $ref: './common.yaml#/components/responses/AuthenticationFailedResponse'
+        '403':
+          $ref: './common.yaml#/components/responses/AuthorizationFailedResponse'
+        '5XX':
+          $ref: './common.yaml#/components/responses/ServerErrorResponse'
+    get:
+      tags: [ Client Configuration ]
+      operationId: getClientConfiguration
+      summary: Get a client's configuration by their ID
+      responses:
+        '200':
+          description: Client exists within the Partner service and their configuration
+            is returned in the response.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GetClientConfigurationResponse'
+        '403':
+          description: Authentication failed.
+        '404':
+          description: Client does not have a configuration recorded within the Partner service.
+        '5XX':
+          description: Server error occurred
+          content:
+            application/json:
+              schema:
+                $ref: './common.yaml#/components/schemas/Error'
+    delete:
+      tags: [ Client Configuration ]
+      operationId: deleteClientConfiguration
+      summary: Delete a client's configuration by their ID
+      responses:
+        '204':
+          description: Client exists within the Partner service and their configuration
+            is returned in the response.
+        '403':
+          description: Authentication failed.
+        '404':
+          description: Client does not have a configuration recorded within the Partner service.
+        '5XX':
+          description: Server error occurred
+          content:
+            application/json:
+              schema:
+                $ref: './common.yaml#/components/schemas/Error'
+  /testConnection:
```

Comment:
According to SAP rules we should not use verbs in context of resource names and "test connection" goes into this direction.